### PR TITLE
feat(gateway): support for ipip-412 parameters

### PR DESCRIPTION
**Pull request by hacdias**
Adds support for IPIP-412 parameters.

- [x] Boxo: https://github.com/ipfs/boxo/pull/370
- [ ] Conformance running against this PR: https://github.com/ipfs/gateway-conformance/pull/87

### CODE REVIEW

File: core/corehttp/gateway.go [hunk 1]
```diff
@@ -156,7 +156,7 @@ func (o *offlineGatewayErrWrapper) ResolvePath(ctx context.Context, path gateway
 	return md, err
 }
 
-func (o *offlineGatewayErrWrapper) GetCAR(ctx context.Context, path gateway.ImmutablePath, params gateway.CarParams) (gateway.ContentPathMetadata, io.ReadCloser, error) {
+func (o *offlineGatewayErrWrapper) GetCAR(ctx context.Context, path gateway.ImmutablePath, params *gateway.CarParams) (gateway.ContentPathMetadata, io.ReadCloser, error) {
```
Comment (Jorropo) [inline core/corehttp/gateway.go:159]: You should add an interface guard.

**Review by Jorropo — CHANGES_REQUESTED** [review]

File: core/corehttp/gateway.go (hunk 1 above)
Comment (hacdias) [reply core/corehttp/gateway.go:159]: @Jorropo hmm, how would that benefit this? The way it is done right now, the code will fail at compile time already.

Comment (Jorropo) [reply core/corehttp/gateway.go:159]: Ah yes NVM thought the interface definition would be here.
Btw you need to be carefull when passing pointers to interfaces, they almost always allocate (they more than likely here).

**Comment by hacdias** [comment]
Merging as per https://github.com/ipfs/boxo/pull/370#pullrequestreview-1537936490 and #9911.

**Pull request merged**